### PR TITLE
Feature/ewl 3984 add documentation

### PR DESCRIPTION
**Pull request by rpama**
## Ticket(s)

**Jira Ticket**

- [EWL-3984: Add Documentation](https://issues.ama-assn.org/browse/EWL-3984)

## Description

Adds a table of contents to the Code Conventions document so people can see what all is in there. Adds description and explanation of pseudo-patterns and their usage.

## To Test

- [x] Go to **docs/code_conventions.md**
- [x] Read

## Relevant Screenshots/GIFs

N/A

## Remaining Tasks

Only if you're unconvinced.

## Additional Notes

No.

### CODE REVIEW

File: styleguide/docs/code_conventions.md [hunk 1]
```diff
@@ -1,7 +1,26 @@
 # Code Conventions
 
+## Contents
```
Comment (chelsiejohnston) [inline styleguide/docs/code_conventions.md]: !!! YAY

**Comment by chelsiejohnston** [comment]
@rpama I just pushed up some edits to the section that talks about the example use cases (mostly to add the example from https://github.com/AmericanMedicalAssociation/AMA-style-guide/pull/229) and hopefully make it easier to read. If my work looks okay to you, feel free to go ahead and merge this! Thanks for getting this down ~on paper~ in `.md`!

bitmoji

File: styleguide/docs/code_conventions.md [hunk 2]
```diff
@@ -10,13 +29,73 @@ You can view both the Twig syntax and the rendered HTML markup for a pattern by
...
 - `?l=3:2|600x400` the label text, with a pipe (|) separating lines. Our images should use [ratio]|[dimensions].
-- `&s=36` the text size. This should only be tweaked if necessary.\
+- `&s=36` the text size. This should only be tweaked if necessary.
 
 ![This is a placeholder image](https://ipsumimage.appspot.com/600x400?l=3:2|600x400&s=36)
 
+### Pseudo-Patterns
+
+[Pseudo-patterns](http://patternlab.io/docs/pattern-pseudo-patterns.html) are Pattern Lab's way of easily managing closely related patterns that have multiple variants. In the case of our style guide, we typically use pseudo-patterns for patterns where the data structures are consistent, but content might be displayed in differently if certain conditions are met. 
+
```
Comment (froboy) [inline styleguide/docs/code_conventions.md]: Only one line break needed here.

File: styleguide/docs/code_conventions.md [hunk 3]
```diff
@@ -10,13 +29,73 @@ You can view both the Twig syntax and the rendered HTML markup for a pattern by
...
+
+Imagine that a CMS serving data to a pattern uses just one data model/content type, but has several different possible ways that content should be displayed. For example:
+* The content must be displayed in different places across the page/site
+* The content must be displayed differently if certain fields are or are not populated
+* The content must be displayed differently depending on whether or not a certain kind of user is logged in
+
+In Drupal, we often use *view modes* to display data in cases like these, so building different view modes in Drupal can often be a good case for using pseudo-patterns in the Style Guide. There are other possible use cases for pseudo-patterns, such as showing pattern states, but this is just the most common example.
```
Comment (froboy) [inline styleguide/docs/code_conventions.md]: I'd recommend "display/view modes" and a link to https://www.drupal.org/docs/8/api/entity-api/display-modes-view-modes-and-form-modes

**Comment by froboy** [comment]
LGTM. Thanks @chelsiejohnston @rpama. Once the conflict is resolved this is good to merge.

**Comment by rpama** [comment]
Conflicts have been resolved. I'm merging!

**Pull request merged**